pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2082,3 +2082,55 @@ def test_merge_equal_cat_dtypes2():
 
     # Categorical is unordered, so don't check ordering.
     tm.assert_frame_equal(result, expected, check_categorical=False)
+
+
+def test_right_merge_preserves_row_order():
+    population = [
+        ("Jenn", "Jamaica", 3),
+        ("Beth", "Bulgaria", 7),
+        ("Carl", "Canada", 30),
+    ]
+    columns = ["name", "country", "population"]
+    pop = DataFrame.from_records(population, columns=columns)
```

Comment:
Shouldn't need `from_records` just use default constructor (applicable in a few places)